### PR TITLE
gNMI-1.10: Update HardwarePort test as per Pull/1318

### PR DESCRIPTION
**Pull request by sgirishj**
Script was expecting specific hardcoded string formation for the below leaf which appears to be vendor specific:
/interfaces/interface/state/hardware-port

Updating it as per https://github.com/openconfig/featureprofiles/pull/1318

Checking for CHASSIS to be an ancestor of a given component -> this step is common between TestHardwarePort and TestComponentParent; hence moved the code to a common func that can be reused between the two tests.

### CODE REVIEW

**Review by greg-dennis — DISMISSED** [review]
Approving for Go quality only

File: feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go [hunk 1]
```diff
@@ -196,16 +196,13 @@ func TestHardwarePort(t *testing.T) {
...
-	// For example, Ethernet3/35/1 hardware port is Ethernet3/35.
-	i := strings.LastIndex(dp.Name(), "/")
-	want := dp.Name()[:i]
-
-	got := gnmi.Get(t, dut, gnmi.OC().Interface(dp.Name()).HardwarePort().State())
-	t.Logf("Got %s HardwarePort from telmetry: %v, expected: %v", dp.Name(), got, want)
-	if got != want {
-		t.Errorf("Get(DUT port1 HardwarePort): got %v, want %v", got, want)
+	// Verify that HardwarePort leaf is present
+	got := gnmi.Lookup(t, dut, gnmi.OC().Interface(dp.Name()).HardwarePort().State())
```
Comment (sachendras) [inline feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:200]: Can we have a Juniper case to address Juniper specific interface naming.

**Comment by sachendras** [comment]
Please modify the test as per https://github.com/openconfig/featureprofiles/pull/1318

File: feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go [hunk 2]
```diff
@@ -211,23 +212,23 @@ func TestHardwarePort(t *testing.T) {
...
+
+	// Verify HardwarePort is a component of type PORT
+	typeGot := gnmi.Get(t, dut, gnmi.OC().Component(val).Type().State())
+	if typeGot != portType {
+		t.Errorf("HardwarePort leaf's component type got %s, want %s", typeGot, portType)
 	}
+
+	// Verify HardwarePort component has CHASSIS as an ancestor
+	verifyChassisIsAncestor(t, dut, val)
+
```
Comment (greg-dennis) [inline feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:231]: delete blank line

File: feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go [hunk 3]
```diff
@@ -211,23 +212,23 @@ func TestHardwarePort(t *testing.T) {
...
-		// Derive hardware port from interface name by removing the port number.
-		// For example, Ethernet3/35/1 hardware port is Ethernet3/35.
-		i := strings.LastIndex(dp.Name(), "/")
-		want = dp.Name()[:i]
-	}
-	got := gnmi.Get(t, dut, gnmi.OC().Interface(dp.Name()).HardwarePort().State())
-	t.Logf("Got %s HardwarePort from telmetry: %v, expected: %v", dp.Name(), got, want)
-	if got != want {
-		t.Errorf("Get(DUT port1 HardwarePort): got %v, want %v", got, want)
+	// Verify HardwarePort leaf is present under interface
```
Comment (greg-dennis) [inline feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:215]: period at end of sentence comment

File: feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go [hunk 4]
```diff
@@ -211,23 +212,23 @@ func TestHardwarePort(t *testing.T) {
...
+	got := gnmi.Lookup(t, dut, gnmi.OC().Interface(dp.Name()).HardwarePort().State())
+	val, present := got.Val()
+	if !present {
+		t.Errorf("DUT port1 %s HardwarePort leaf not found", dp.Name())
+	}
+	t.Logf("For interface %s, HardwarePort is %s", dp.Name(), val)
+
+	// Verify HardwarePort is a component of type PORT
```
Comment (greg-dennis) [inline feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:223]: period at end of sentence comment

File: feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go [hunk 5]
```diff
@@ -211,23 +212,23 @@ func TestHardwarePort(t *testing.T) {
...
+	typeGot := gnmi.Get(t, dut, gnmi.OC().Component(val).Type().State())
+	if typeGot != portType {
+		t.Errorf("HardwarePort leaf's component type got %s, want %s", typeGot, portType)
 	}
+
+	// Verify HardwarePort component has CHASSIS as an ancestor
```
Comment (greg-dennis) [inline feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:229]: period at end of sentence comment

File: feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go [hunk 6]
```diff
@@ -379,6 +380,32 @@ func findComponentsListByType(t *testing.T, dut *ondatra.DUTDevice) map[string][
 	return s
 }
 
+// verifyChassisIsAncestor verifies that a given component has
+// a component of type CHASSIS as an ancestor
```
Comment (greg-dennis) [inline feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:384]: period at end of sentence comment

**Review by greg-dennis — CHANGES_REQUESTED** [review]

**Review by greg-dennis — APPROVED** [review]
Approving for Go quality only

**Review by mojiiba — APPROVED** [review]

**Pull request merged**